packages/app-cli/app/gui/FolderListWidget.ts

```diff
@@ -4,12 +4,97 @@ import BaseModel from '@joplin/lib/BaseModel';
 import Setting from '@joplin/lib/models/Setting';
 import { _ } from '@joplin/lib/locale';
 import { FolderEntity } from '@joplin/lib/services/database/types';
-import { getDisplayParentId, getTrashFolderId } from '@joplin/lib/services/trash';
+import {
+	getDisplayParentId,
+	getTrashFolderId,
+} from '@joplin/lib/services/trash';
 const ListWidget = require('tkwidgets/ListWidget.js');
 
-export default class FolderListWidget extends ListWidget {
+// Service for managing folder collapse states
+class FolderCollapseService {
+	private static instance_: FolderCollapseService;
+	private collapsedFolders_: Set<string> = new Set();
+
+	public static instance(): FolderCollapseService {
+		if (!this.instance_) {
+			this.instance_ = new FolderCollapseService();
+			this.instance_.loadFromSettings();
+		}
+		return this.instance_;
+	}
+
+	public isCollapsed(folderId: string): boolean {
+		return this.collapsedFolders_.has(folderId);
+	}
+
+	public setCollapsed(folderId: string, collapsed: boolean) {
+		if (collapsed) {
+			this.collapsedFolders_.add(folderId);
+		} else {
+			this.collapsedFolders_.delete(folderId);
+		}
+		this.saveToSettings();
+	}
+
+	public toggleCollapsed(folderId: string): boolean {
+		const newState = !this.isCollapsed(folderId);
+		this.setCollapsed(folderId, newState);
+		return newState;
+	}
+
+	public expandToFolder(folderId: string, folders: FolderEntity[]) {
+		// Find all parent folders and expand them
+		const parentsToExpand: string[] = [];
+		let currentId = folderId;
+
+		while (currentId) {
+			const folder = BaseModel.byId(folders, currentId);
+			if (!folder) break;
+
+			const parentId = getDisplayParentId(
+				folder,
+				folders.find((f) => f.id === folder.parent_id),
+			);
+			if (parentId) {
+				parentsToExpand.unshift(parentId);
+				currentId = parentId;
+			} else {
+				break;
+			}
+		}
+
+		// Expand all parent folders
+		for (const parentId of parentsToExpand) {
+			this.setCollapsed(parentId, false);
+		}
+	}
+
+	private loadFromSettings() {
+		try {
+			const stored = Setting.value('cli.folderCollapseState');
+			if (stored && typeof stored === 'string') {
+				const folderIds = JSON.parse(stored);
+				this.collapsedFolders_ = new Set(folderIds);
+			}
+		} catch (error) {
+			// If there's an error loading, start with empty state
+			this.collapsedFolders_ = new Set();
+		}
+	}
+
+	private saveToSettings() {
+		try {
+			const folderIds = Array.from(this.collapsedFolders_);
+			Setting.setValue('cli.folderCollapseState', JSON.stringify(folderIds));
+		} catch (error) {
+			// Silently ignore save errors
```

Comment:
Please don't ignore errors - our philosophy is to let things crash if there's an error. It's much easier to manage over the long term since there will be less hidden errors

Reply:
Ok.